libs/api/books/src/lib/reading-list.service.ts

```diff
@@ -28,4 +28,10 @@ export class ReadingListService {
       return list.filter(x => x.bookId !== id);
     });
   }
-}
+  
+  async markBookAsFinished(id: string, finishDate: string) {
+    this.storage.update(list => list.filter(listItem =>
```

Comment:
Change this to `list.map` so the book is saved properly. If the page is reloaded and the booklist read from storage, the book finished date isn't returned properly.

Reply:
Resolved